Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,341 @@
+            ValidateObjects(keyValuePairs, keyValueValidationFunctions, validationFeedback, syntaxConf);
+            List<ValidationStruct> structuredEntries = BuildValidationStructs(keyValuePairs, syntaxConf);
+            ValidateObjects(structuredEntries, structuredValidationFunctions, validationFeedback, syntaxConf);
+
+            return new([.. validationFeedback], structuredEntries);
+        }
+
+        /// <summary>
+        /// Gets the encoding of a PX file stream
+        /// </summary>
+        /// <param name="stream">The stream to be validated</param>
+        /// <param name="syntaxConf">Syntax configuration object</param>
+        /// <param name="validationFeedback">List of reports that stores any issues that were ran into during validation process</param>
+        /// <param name="filename">Name of the PX file being validated</param>
+        /// <returns>Character encoding if found</returns>
+        public static Encoding? GetEncoding(Stream stream, PxFileSyntaxConf syntaxConf)
+        {
+            try
+            {
+                return PxFileMetadataReader.GetEncoding(stream, syntaxConf);
+            }
+            catch (InvalidPxFileMetadataException)
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously validates the syntax of a PX file's metadata.
+        /// </summary>
+        /// <param name="stream">The stream of the PX file to be validated.</param>
+        /// <param name="encoding">The encoding format to use for the PX file reading</param>
+        /// <param name="filename">The name of the file to be validated.</param>
+        /// <param name="syntaxConf">An optional <see cref="PxFileSyntaxConf"/> parameter that specifies the syntax configuration for the PX file. If not provided, the default syntax configuration is used.</param>
+        /// <param name="bufferSize">An optional parameter that specifies the buffer size for reading the file. If not provided, a default buffer size of 4096 is used.</param>
+        /// <param name="customValidationFunctions">An optional <see cref="CustomValidationFunctions"/> parameter that specifies custom validation functions to be used during validation. If not provided, the default validation functions are used.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> parameter that can be used to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous operation. The task result contains a <see cref="SyntaxValidationResult"/> object which includes a <see cref="ValidationReport"/> and a list of <see cref="ValidationStruct"/> objects. The ValidationReport contains feedback items that provide information about any syntax errors or warnings found during validation. The list of ValidationStruct objects represents the structured objects in the PX file that were validated.</returns>
```

Comment:
The task structure can be seen from the return type and is redundant to document. The rest is good.